app/src/main/java/com/example/itbook/ui/detailbook/DetailBookFragment.kt

```diff
@@ -0,0 +1,165 @@
+import kotlinx.android.synthetic.main.preview_category.view.*
+import org.json.JSONException
+
+class DetailBookFragment : BaseFragment(), DetailBookContract.View {
+    override val layoutResource: Int = R.layout.fragment_detail_book
+
+
+    private var presenter: DetailBookPresenter? = null
+    private var loadingDialogFragment: LoadingDialogFragment? = null
+    private val bookAdapter = BookAdapter(itemClickListener = this::onItemBookClickListener)
+    private var books = mutableListOf<Book>()
+    private var book: Book? = null
+    private var isFavoriteBook = false
+    private var isLocalBook = false
+
+    override fun initViews() {
+        val titleSimilarBook = resources.getString(R.string.similar_book)
+        layoutPreviewCategory.apply {
+            recyclerViewBooks.adapter = bookAdapter
+            textPreviewCategory.text = titleSimilarBook
+        }
+        loadingDialogFragment = LoadingDialogFragment()
+        loadingDialogFragment?.isCancelable = false
+        isSetup = false
+    }
+
+    override fun initData() {
+        var isbn13: String? = null//id to search book
+        arguments?.let {
+            isLocalBook = it.getBoolean(STRING_LOCAL, false)
+            isbn13 = it.getString(Book.ISBN13)
+        }
+        activity?.let {
+            val bookRepository = BooksRepository.getInstance(
+                BooksRemoteDataSource.getInstance(BookRemoteHandler()),
+                BooksLocalDataSource.getInstance(BookDaoImp(BooksDatabase.getInstance(it)))
+            )
+            presenter =
+                isbn13?.let { it1 -> DetailBookPresenter(this, bookRepository, it1, isLocalBook) }
+            presenter?.start()
+        }
+
+    }
+
+    override fun initListeners() {
+        imageFavoriteDetailBook.setOnClickListener { onFavoriteClickListener() }
+        imageBackDetailBook.setOnClickListener { onBackClickListener() }
+        imageShareDetailBook.setOnClickListener { onShareClickListener() }
+        layoutPreviewCategory.textPreviewCategory.setOnClickListener { onSimilarBooksClickListener() }
+    }
+
+    override fun showBook(book: Book) {
+        book.apply {
+            imageDetailBook.loadImageFromUri(image)
+            textTitleDetailBook.text = title
+            textAuthorDetailBook.text = authors
+            textDescDetailBook.text = desc
+            imageRatingDetailBook.setImageResource(getImageRating(rating))
+            textPublisherDetailBook.text = publisher
+            textLanguageDetailBook.text = language
+            textPapersDetailBook.text = pages.toString()
+            textYearDetailBook.text = year.toString()
+            textIsbn10DetailBook.text = isbn10
+            textIsbn13DetailBook.text = isbn13
+            LinkBuilder().build(textLinkDetailBook, url)
+        }
+        this.book = book
+    }
+
+    override fun showSimilarBooks(books: List<Book>) {
+        bookAdapter.updateData(books.toList())
+        this.books = books.toMutableList()
```

Comment:
thằng this.books này dùng để làm gì?

Reply:
thằng này để khi mình chọn book trong similar book thì nó sẽ lấy id của book trong books để hiển thị trang detail book mới ạ